From 2510308490b642424826d349c9856801c8b45ac0 Mon Sep 17 00:00:00 2001
From: Conn O'Griofa <connogriofa@gmail.com>
Date: Sat, 18 Jan 2014 18:39:52 +0000
Subject: [PATCH] Implement BOARD_CAMERA_NO_UNWANTED_MSG

On msm7x27 devices, most camera hals appear to spam this message:

lockIfMessageWanted(16): dropped unwanted message

This happens for all camera modes except barcode scanning. If there
is a better fix for this issue, please let me know.

Change-Id: I266c6d21196c193e389e6773d532591df7b3534a
---
 services/camera/libcameraservice/Android.mk            | 4 ++++
 services/camera/libcameraservice/api1/CameraClient.cpp | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/services/camera/libcameraservice/Android.mk b/services/camera/libcameraservice/Android.mk
index b04c27c..b941b92 100644
--- a/services/camera/libcameraservice/Android.mk
+++ b/services/camera/libcameraservice/Android.mk
@@ -78,6 +78,10 @@ ifeq ($(BOARD_CAMERA_MSG_MGMT),true)
     LOCAL_CFLAGS += -DCAMERA_MSG_MGMT
 endif
 
+ifeq ($(BOARD_CAMERA_NO_UNWANTED_MSG),true)
+    LOCAL_CFLAGS += -DCAMERA_NO_UNWANTED_MSG
+endif
+
 LOCAL_MODULE:= libcameraservice
 
 include $(BUILD_SHARED_LIBRARY)
diff --git a/services/camera/libcameraservice/api1/CameraClient.cpp b/services/camera/libcameraservice/api1/CameraClient.cpp
index afaf12a..8d6f75c 100644
--- a/services/camera/libcameraservice/api1/CameraClient.cpp
+++ b/services/camera/libcameraservice/api1/CameraClient.cpp
@@ -763,7 +763,9 @@ bool CameraClient::lockIfMessageWanted(int32_t msgType) {
         }
         usleep(CHECK_MESSAGE_INTERVAL * 1000);
     }
+#ifndef CAMERA_NO_UNWANTED_MSG
     ALOGW("lockIfMessageWanted(%d): dropped unwanted message", msgType);
+#endif
     return false;
 }
 
-- 
1.8.3.2

